src/EncompassRest/LoanPipeline/LoanPipelineCursor.cs

```
@@ -23,59 +23,53 @@ internal LoanPipelineCursor(EncompassRestClient client, string cursorId, int cou
+            //Preconditions.LessThanOrEquals(start, nameof(start) , Count, nameof(Count));
+            
 
-            return GetItemsInternalAsync(start, count, cancellationToken);
+            return GetItemsInternalAsync(start, count,itemFields, cancellationToken);
         }
 
-        private async Task<List<LoanPipelineData>> GetItemsInternalAsync(int start, int count, CancellationToken cancellationToken)
+        private async Task<List<LoanPipelineData>> GetItemsInternalAsync(int start, int count, IEnumerable<string> itemFields, CancellationToken cancellationToken)
         {
-            List<LoanPipelineData> data = null;
-            do
+            var retrievedData = await Client.Pipeline.ViewPipelineCursorInternalAsync(CursorId, null, itemFields == null?Fields:itemFields, start, count, cancellationToken, nameof(GetItemAsync), async response =>
             {
-                var retrievedCount = data?.Count ?? 0;
-                var retrievedData = await Client.Pipeline.ViewPipelineCursorInternalAsync(CursorId, null, Fields, start + retrievedCount, count - retrievedCount, cancellationToken, nameof(GetItemAsync), async response =>
-                {
-                    var list = await response.Content.ReadAsAsync<List<LoanPipelineData>>().ConfigureAwait(false);
-                    if (list.Count == 0)
-                    {
-                        throw await RestException.CreateAsync("Failed to retrieve pipeline data", response).ConfigureAwait(false);
-                    }
-                    return list;
-                }).ConfigureAwait(false);
-                if (data == null)
-                {
-                    data = retrievedData;
-                }
-                else
+                var list = await response.Content.ReadAsAsync<List<LoanPipelineData>>().ConfigureAwait(false);
+                if (list.Count == 0)
```

No longer need to validate the returned count.

---

not sure about this one.  should we return an empty list? or let it throw the error?

---

I don't foresee the API ever returning an empty list.

---

hmm....ok.  remove.